### PR TITLE
First sound test

### PR DESCRIPTION
**Pull request by daschuer**
In an attempt to fix https://github.com/mixxxdj/mixxx/issues/11094 I have created a test that compares the first sound position of the current version with the position of future versions. 

The idea was to replace the faulty 10.0.20348.1 Media Foundation version with FFmpeg. 
This test exposes that this is not trivial, because the Media Foundation has an offset compared to FFmpeg according to a not yet known pattern. 

We need further investigations to either fix https://github.com/mixxxdj/mixxx/issues/11094 with Media Foundation or mimic the offset behaviour with FFmpeg. The alternative would be to just swap to FFmpeg and tell the users to reanalyse the track and adust the cue points again.  

In addition this has discovered a regression in the 2.3 branch where the `FPM_64BIT` has been lost: 
https://github.com/mixxxdj/buildserver/blob/2.3.x-windows/build/libmad-0.15.1b/msvc%2B%2B/libmad.vcxproj#L133
This falls back to 32 bit maths leading to significant rounding issues.  
https://github.com/markjeee/libmad/blob/c2f96fa4166446ac99449bdf6905f4218fb7d6b5/fixed.h#L425
`This version is the most portable but it loses significant accuracy.`

This PR is just a test that aims to detect offset issues due to decoder update early and should be merged independently.

### CODE REVIEW

**Comment by daschuer** [comment]
The new clang format insist to break the format. I suggest to just merge it as it is. I have no interest to debug it, since we run clang  format on changes only.

**Comment by daschuer** [comment]
Rebasing to the updated updated pre-commit does not help. I give up.

**Comment by daschuer** [comment]
I have no interest to maintain clang-format exceptions because they break nice formatted comments as well. 
I just when ahead and applied the clang-format suggestions.

**Comment by daschuer** [comment]
This is now adjusted for the FPM_64BIT libmad version and merged from 2.4. 
Ready for review and merge.

File: src/sources/soundsourcemp3.cpp [hunk 1]
```diff
@@ -680,6 +685,14 @@ ReadableSampleFrames SoundSourceMp3::readSampleFramesClamped(
                                     << "Recoverable MP3 frame decoding error:"
                                     << mad_stream_errorstr(&m_madStream);
                         }
+                    } else if (m_madStream.error == MAD_ERROR_BADDATAPTR &&
+                            m_curFrameIndex == firstFrameIndex) {
+                        // This is expected after starting decoding with an offset
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/sources/soundsourcemp3.cpp:690]: I have debugged into that and there is is nothing wrong. That why I have silenced the message. 
You find this in other adoptions as well. Here for instance: 
https://github.com/Sneeds-Feed-and-Seed/sneedacity/blob/5fba67545ca29d009b51d355687979a3996d252c/src/import/ImportMP3.cpp#L1093

File: src/test/soundproxy_test.cpp [hunk 2]
```diff
@@ -747,6 +748,128 @@ TEST_F(SoundSourceProxyTest, regressionTestCachingReaderChunkJumpForward) {
...
+            // https://github.com/mixxxdj/mixxx/issues/11888
+            // 1166 FFmpeg
+
+            {QStringLiteral("cover-test.ogg"), 1166},
+            {QStringLiteral("cover-test.opus"), 1268},
+            {QStringLiteral("cover-test.wav"), 1166},
+            {QStringLiteral("cover-test.wav"), 1166},
+            {QStringLiteral("cover-test.wv"), 1166}};
+
+    for (const auto& ref : refs) {
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/test/soundproxy_test.cpp:814]: The test already continues when one test is failing. I am fine with the current implementation.

File: src/sources/soundsourcemediafoundation.cpp [hunk 3]
```diff
@@ -86,11 +86,11 @@ QString SoundSourceProviderMediaFoundation::getVersionString() const {
...
-            return QStringLiteral("%1.%2.%3.%4")
-                    .arg(HIWORD(pVerInfo->dwProductVersionMS))
-                    .arg(LOWORD(pVerInfo->dwProductVersionMS))
-                    .arg(HIWORD(pVerInfo->dwProductVersionLS))
-                    .arg(LOWORD(pVerInfo->dwProductVersionLS));
+            return QStringLiteral("%1.%2.%3.%4").arg(
+                    QString::number(HIWORD(pVerInfo->dwProductVersionMS)),
+                    QString::number(LOWORD(pVerInfo->dwProductVersionMS)),
+                    QString::number(HIWORD(pVerInfo->dwProductVersionLS)),
+                    QString::number(LOWORD(pVerInfo->dwProductVersionLS)));
```
Comment (Swiftb0y) [inline src/sources/soundsourcemediafoundation.cpp:93]: Really looking forward to Qt6 when we finally have a variadic `QString::arg`...

**Comment by daschuer** [comment]
The failing test confirms that the Media Foundation version is still read: 
`D:/a/mixxx/mixxx/src/test/id3-test-data/cover-test-ffmpeg-aac.m4a Microsoft Media Foundation 10.0.17763.2989`
https://github.com/mixxxdj/mixxx/actions/runs/6077689437/job/16487777138?pr=11887

**Comment by daschuer** [comment]
All green now.

**Review by Swiftb0y — APPROVED** [review]
any1 else wanting to take a look?

**Review by JoergAtGithub — APPROVED** [review]
LGTM!

**Comment by JoergAtGithub** [comment]
Thank you!

**Pull request merged**